### ISSUE TITLE
Capability: Embed an interactive map viewer, using HTML markup

### ISSUE THREAD

**Issue by Malvoz** [Maps4HTML/HTML-Map-Element-UseCases-Requirements#137]
This issue is for discussion of <a href="https://maps4html.github.io/HTML-Map-Element-UseCases-Requirements/#capability-declarative-map">the required map viewer capability “Embed an interactive map viewer, using HTML markup”</a>.

**Comment by Malvoz** [comment]
@AmeliaBR, @prushforth 

Parallels have been drawn between the map viewer model and `<iframe>`. In relation, there is a proposal for the [`<portal>`](https://github.com/WICG/portals) element which is receiving some push-back in this [TAG design review (comment)](https://github.com/w3ctag/design-reviews/issues/331#issuecomment-529805604), and so I wonder if this could indicate implications or otherwise affect the establishment of a model for the map viewer?

**Comment by prushforth** [comment]
Good question, answer TBD. I think a map viewer that allows composition / layering across origins will require us to come up with a good story. Hopefully we can get some contributions, or at least inform our own contributions from the experience and designs from people with experience in designing similar things in iframe and portal contexts. Portals and iframes are more generic containers than maps, so I believe our biggest challenge will be selling a highly specific type of container, like a map viewer. 

All the great work this community is currently doing will be our business case!

**Comment by AmeliaBR** [comment]
Yes, one of the more difficult things to spec will be the model for creating fully interactive, style-able layers from external files. Another relevant model from that perspective is the SVG `<use>` element (which is currently restricted to same-domain external files, but there is discussion in w3c/svgwg#707 about extending it to cross-origin content).

**Comment by prushforth** [comment]
If I understand [this](https://github.com/w3c/svgwg/issues/707#issuecomment-522351290) correctly, layers in a map viewer, were they to be loaded via a URL (as in layer src), would have to be their own browsing context, if they were to be able to load resources, such as tiles or WMS/image requests (which is pretty important for map layers).

**Comment by prushforth** [comment]
I think this issue is about how an HTML author would go about adding a map viewer to their page.  I think they should be able to do so by adding an element, in an analagous manner the method they can use to embed `<video>` players.

**Comment by prushforth** [comment]
This comment doesn't apply to the [PR](https://github.com/Maps4HTML/HTML-Map-Element-UseCases-Requirements/pull/237), but to the text about the existing implementations.

Not to be too hard on the -embed list, but they are all JavaScript API dependent, too.  The commentary seems to imply that the JavaScript dependency only applies to custom element wrappers of mapping libraries, and that the embeds are declarative except for the fact that they rely on configuration in the source URL. In fact, ALL the embeds are entirely animated by their associated JavaScript APIs internally.

**Comment by Malvoz** [comment]
https://github.com/Maps4HTML/HTML-Map-Element-UseCases-Requirements/issues/137#issuecomment-721980914

> Not to be too hard on the -embed list, but they are all JavaScript API dependent

Yes but the capability is about embedding a web map with HTML code alone, which holds true from the point of view of the HTML author (so separate from [Capability: Display a basic map without JavaScript](https://maps4html.org/HTML-Map-Element-UseCases-Requirements/#capability-no-js)).
